server/images.go

```diff
@@ -300,250 +263,197 @@ func CreateModel(ctx context.Context, name, modelFileDir, quantization string, c
 		},
 	}
 
-	var layers Layers
-	messages := []string{}
-
-	params := make(map[string][]string)
-	fromParams := make(map[string]any)
+	var messages []*api.Message
+	parameters := make(map[string]any)
 
+	var layers []*Layer
 	for _, c := range commands {
 		mediatype := fmt.Sprintf("application/vnd.ollama.image.%s", c.Name)
 
 		switch c.Name {
-		case "model":
-			if strings.HasPrefix(c.Args, "@") {
-				blobPath, err := GetBlobsPath(strings.TrimPrefix(c.Args, "@"))
+		case "model", "adapter":
+			var layers2 *ordered.Map[*Layer, *llm.GGML]
```

Comment:
```suggestion
			var layers2 *ordered.Map[*Layer, *llm.GGML] // layers2 stores the layers of an existing parent model
```
Would this comment be correct?

Reply:
I've updated the var name so this should be clearer